x/mongo/driver/operation/find.toml

```diff
@@ -90,6 +90,11 @@ documentation = "AwaitData when true makes a cursor block before returning when
 type = "boolean"
 documentation = "AllowPartialResults when true allows partial results to be returned if some shards are down."
 
+[request.allowDiskUse]
+type = "boolean"
+minWireVersion = 4
+documentation = "AllowDiskUse when true allows temporary data to be written to disk during the find command. Valid for server versions >= 3.2. For servers < 3.2, this setting is ignored."
```

Comment:
It's only valid for MongoDB 4.4 and up.  It's just that the server will return an error if it's sent as a field in the find command (version 3.2 and up).  The reason the spec says anything about server versions < 3.2 is that there's no way to send this field in the old protocol, so the driver ignores it rather than throws a client side error.

Reply:
@jyemin Your comment implies that this should check for wire version 9. Is this correct? The spec doesn't say anything about client-side errors, so I don't see why any wire version checking is needed here. If a user sends it to a <4.4 server, the server will error.

Reply:
My comment was more about the documentation.  I don't see any client-side checks in the generated code.  What is the actual effect of minWireVersion?

Reply:
I think the actual tag is `minWireVersionRequired`. It's used for collation (https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/operation/find.toml#L93) and the generated code is https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/operation/find.go#L116.

Reply:
because we don't want it to error, just to ignore it, I used the minWireVersion tag instead: https://github.com/mongodb/mongo-go-driver/blob/feb20c34fd657c992f500c005e7f84b0cb4735e7/x/mongo/driver/operation/insert.toml#L34

Reply:
Ah, I see what this is doing now. I don't think this is quite in line with the spec, though, which says that this is not sent for versions < 3.2. So I think you want the wire version check from the previous commit (minWireVersion=4) and the documentation from the new commit. The documentation should probably be amended to say that the server will error for versions < 4.4.

@jyemin Am I understanding this correctly? This is the first version-dependent option we've added in a while AFAIK so I'm not sure what the precedent is.

Reply:
We want the behavior to be that:

1. allowDiskUse is ignored for the old protocol (pre-find command OP_QUERY)
2. allowDiskUse is honored for the new protocol (find command) regardless of server version

IIRC the code generation doesn't handle case 1 at all, and case 2 doesn't need a version check.  Is that correct?

Reply:
The legacy code path goes through the operation's `command` function and then goes through the created document and translates each element. Having the `minWireVersion` check is fine because it just prevents the `allowDiskUse` element from being added in the `command` function, but it's not explicitly needed.

@iwysiu The new documentation needs to be changed to say the server will error rather than the driver will error.